numpy/ma/core.py

```diff
@@ -3183,29 +3185,80 @@ def __getitem__(self, indx):
         Return the item described by i, as a masked array.
 
         """
-        dout = self.data[indx]
         # We could directly use ndarray.__getitem__ on self.
         # But then we would have to modify __array_finalize__ to prevent the
         # mask of being reshaped if it hasn't been set up properly yet
         # So it's easier to stick to the current version
+        dout = self.data[indx]
         _mask = self._mask
+
+        def _is_scalar(m):
```

Comment:
To me, it is not all that helpful to have these as helper functions, and I wonder a bit about the additional hit in performance.